scripts/Deployment.s.sol

```diff
@@ -0,0 +1,139 @@
+
+        // Pair factory
+        PairFactory pairFactory = new PairFactory();
+
+        // Router
+        Router router = new Router(address(pairFactory), WCANTO);
+
+        // VelocimeterLibrary
+        VelocimeterLibrary velocimeterLib = new VelocimeterLibrary(address(router));
+
+        // VeArtProxy
+        VeArtProxy veArtProxy = new VeArtProxy();
+
+        // VotingEscrow
+        VotingEscrow votingEscrow = new VotingEscrow(address(flow), address(veArtProxy), TEAM_MULTI_SIG);
+
+        // RewardsDistributor
+        RewardsDistributor rewardsDistributor = new RewardsDistributor(address(votingEscrow));
+
+        // Wrapped external bribe factory
+        WrappedExternalBribeFactory wrappedExternalBribeFactory = new WrappedExternalBribeFactory();
+
+        // Voter
+        Voter voter = new Voter(
+            address(votingEscrow),
+            address(pairFactory),
+            address(gaugeFactory),
+            address(bribeFactory),
+            address(wrappedExternalBribeFactory)
+        );
+
+        // Set voter
+        wrappedExternalBribeFactory.setVoter(address(voter));
+        votingEscrow.setVoter(address(voter));
+        pairFactory.setVoter(address(voter));
+
+        // Minter
+        Minter minter = new Minter(
+            address(voter),
+            address(votingEscrow),
+            address(rewardsDistributor)
+        );
+
+        // Set flow minter to contract
+        flow.setMinter(address(minter));
+
+        // Set pair factory pauser and tank
+        pairFactory.setPauser(TEAM_MULTI_SIG);
+        pairFactory.setTank(TANK);
+
+        // Set voting escrow's voter and art proxy
+        votingEscrow.setVoter(address(voter));
+        votingEscrow.setArtProxy(address(veArtProxy));
+
+        // Set minter and voting escrow's team
+        votingEscrow.setTeam(TEAM_MULTI_SIG);
+        minter.setTeam(TEAM_MULTI_SIG);
+        pairFactory.setTeam(TEAM_MULTI_SIG);
+
+        // Set fee manager
+        pairFactory.setFeeManager(TEAM_MULTI_SIG);
+
+        // Set voter's governor
+        voter.setGovernor(TEAM_MULTI_SIG);
+
+        // Set voter's emergency council
+        voter.setEmergencyCouncil(TEAM_MULTI_SIG);
+
+        // Set rewards distributor's depositor to minter contract
+        rewardsDistributor.setDepositor(address(minter));
+
+        // Initialize tokens for voter
+        address[] memory whitelistedTokens = new address[](19);
+        whitelistedTokens[0] = address(flow);
+        whitelistedTokens[1] = 0x4e71a2e537b7f9d9413d3991d37958c0b5e1e503; // NOTE
+        whitelistedTokens[2] = 0x80b5a32e4f032b2a058b4f29ec95eefeeb87adcd; // USDC
+        whitelistedTokens[3] = 0x5db67696c3c088dfbf588d3dd849f44266ff0ffa; // CRE
+        whitelistedTokens[4] = WCANTO;
+        whitelistedTokens[5] = 0xeceeefcee421d8062ef8d6b4d814efe4dc898265; // ATOM
+        whitelistedTokens[6] = 0x1d54ecb8583ca25895c512a8308389ffd581f9c9; // INJ
```

Comment:
I would reccomend  not whitelisting so many tokens and using the governor method to create gauges. Otherwise we will have people creating loads of shitty pools. No big deal it just makes a huge mess thats all.